### PR TITLE
Update README.md page with documentation.

### PR DESCRIPTION
**Pull request by cschuet**
[Rendered link](https://github.com/cschuet/async_grpc/blob/b14cf00081c2133d0542562c7d2686939fbb92f0/README.md)

### CODE REVIEW

**Comment by cschuet** [comment]
@pifon2a I also have usage example, but I will add those in a later commit.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,150 @@
...
+Fortunately gRPC offers a way to asynchronously schedule RPC actions, e.g. RequestAsyncRead(), and also an event queue mechanism to wait for the completion of these events, e.g. ::grpc::CompletionQueue.
+
+So we built a small server framework on top of the existing gRPC async mechanisms to make it convenient to write asynchronous gRPC servers.
+
+## Overview
+<a name="overview"></a>
+
+<div style="text-align:center"><img src ="gfx/diagram.png" /></div>
+
+<div style="text-align:center">Figure 1: gRPC framework architecture overview</div>
```
Comment (pifon2a) [inline README.md]: nit: period after 'overview'.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,150 @@
...
+
+The above class diagram provides an overview of the most important concepts in the framework.
+Let's look at these in turn.
+
+
+### Server
+<a name="server"></a>
+
+The `Server` class owns all server-wide resources, such as the gRPC completion queues, all event queues and the threads that process these queues (see [Threading Model](#threading-model) section below for more details).
+In addition to that the server may own an implementation of `ExecutionContext` which provides shared state that can be accessed by `RpcHandler`s in a thread-safe manner (see [here](#example-sharing-resources-across-handler-invocations) for example).
```
Comment (pifon2a) [inline README.md]: `RpcHandler`s looks a bit weird when rendered: maybe just `RpcHandler` or `RpcHandlers?`

File: README.md [hunk 3]
```diff
@@ -1,2 +1,150 @@
...
+The `ActiveRpcs` object owned by the services is responsible for that. An `EventQueueSelector` is used to select an event queue for incoming requests.
+By default we simply assign them in a round-robin fashion.
+
+
+### Rpc
+<a name="rpc"></a>
+
+An `Rpc` represents a connection from a particular client for a specific method invocations.
+One client invoking two different methods on the same server is therefore represented by two `Rpc` objects.
+`N * M Rpc` objects are created when the server is started waiting for incoming client connections, where `N` is the number of completion queues and `M` is the total number of methods offered by the server across all services.
```
Comment (pifon2a) [inline README.md]: nit: `N * M` `Rpc`

**Review by pifon2a — APPROVED** [review]
Thanks for documenting this!

**Pull request merged**